### PR TITLE
feat(notifications): Create Subscription DTO and Model

### LINKED ISSUE #431: [V2 notifications] Create Subscription DTO and Model
Create Subscription DTO and Model for V2 API

### PR DESCRIPTION
**Pull request by FelixTing**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/go-mod-core-contracts/blob/master/.github/Contributing.md.

## What is the new behavior?
Create Subscription DTO and Model for Notifications V2 API

Close #431 

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## New Imports

- [ ] Yes
- [x] No

## Specific Instructions
Are there any specific instructions or things that should be known prior to reviewing?

## Other information

### CODE REVIEW

**Comment by FelixTing** [comment]
Note that this PR should be on-hold per Core WG decision to revisit the Support service API design before moving forward with implementations.

**Review by cloudxxx8 — CHANGES_REQUESTED** [review]
please make the change to match the latest swagger modifications
for example, adding resendLimit and resendInterval
https://github.com/edgexfoundry/edgex-go/pull/3030/files
https://app.swaggerhub.com/apis-docs/EdgeXFoundry1/support-notifications/2.x#/Subscription

**Comment by FelixTing** [comment]
recheck

File: v2/dtos/requests/subscription.go [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/v2"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/v2/dtos"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/v2/dtos/common"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/v2/models"
+)
+
+// AddSubscriptionRequest defines the Request Content for POST Subscription DTO.
+// This object and its properties correspond to the AddSubscriptionRequest object in the APIv2 specification:
+// https://app.swaggerhub.com/apis-docs/EdgeXFoundry1/support-notifications/2.x#/AddSubscriptionRequest
+type AddSubscriptionRequest struct {
```
Comment (lenny-intel) [inline v2/dtos/requests/subscription.go:21]: Please add factory method that takes parameters for the required properties on the Subscription DTO. Factory method should auto set properties like `ApiVersion`, `RequestId`, etc.
See this PR for example:
https://github.com/edgexfoundry/go-mod-core-contracts/pull/468

File: v2/dtos/requests/subscription.go [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+		d := dtos.ToSubscriptionModel(req.Subscription)
+		s = append(s, d)
+	}
+	return s
+}
+
+// UpdateSubscriptionRequest defines the Request Content for PUT event as pushed DTO.
+// This object and its properties correspond to the UpdateSubscriptionRequest object in the APIv2 specification:
+// https://app.swaggerhub.com/apis-docs/EdgeXFoundry1/support-notifications/2.x#/UpdateSubscriptionRequest
+type UpdateSubscriptionRequest struct {
```
Comment (lenny-intel) [inline v2/dtos/requests/subscription.go:63]: Add factory method. In this case all is optional, so no parameters, but should auto set properties that can be set.

**Review by lenny-intel — CHANGES_REQUESTED** [review]

**Comment by FelixTing** [comment]
Considering the example PR #468 is still in progress and the discussion of the factory method in PR #462 was inconclusive, factory methods will be added by another PR afterwards.

**Comment by FelixTing** [comment]
rebased

**Review by lenny-intel — APPROVED** [review]

File: v2/dtos/requests/subscription_test.go [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+	testSubscriptionResendLimit    = int64(5)
+	testSubscriptionResendInterval = "10s"
+	unsupportedChannelType         = "unsupportedChannelType"
+	unsupportedCategory            = "unsupportedCategory"
+)
+
+func addSubscriptionRequestData() AddSubscriptionRequest {
+	return AddSubscriptionRequest{
+		BaseRequest: common.BaseRequest{
+			RequestId: ExampleUUID,
```
Comment (lenny-intel) [inline v2/dtos/requests/subscription_test.go:38]: Add setting of ApiVersion now that is is required. Example here:
https://github.com/edgexfoundry/go-mod-core-contracts/blob/master/v2/dtos/requests/device_test.go#L40

**Review by lenny-intel — CHANGES_REQUESTED** [review]

File: v2/dtos/requests/subscription_test.go (hunk 3 above)
Comment (FelixTing) [reply v2/dtos/requests/subscription_test.go]: Added.

**Review by lenny-intel — APPROVED** [review]
LGTM

**Pull request merged**